config/300-azureblobstoragesource.yaml

```diff
@@ -110,6 +98,31 @@ spec:
                   # Allowed for completeness, but users should be aware that those events use a different schema from
                   # the other Storage event types.
                   - Microsoft.Storage.BlobInventoryPolicyCompleted
+              endpoint:
+                description: The destination of events subscribed via Event Grid.
+                type: object
+                properties:
+                  eventHubs:
```

Comment:
IIRC, Azure does support multiple event hubs within an eventhub namespace, but given the use case here, shouldn't it be singular?

Reply:
I never know whether this should be singular or plural!

The intended meaning here was `destination: <name of an Azure service>`.
Azure refers to their _service_ as "Event Hubs" (plural). A significant example from their docs

> An Event Hubs namespace is a management container ...
>
> -- https://docs.microsoft.com/en-us/azure/event-hubs/event-hubs-features

They seem to use "an Event Hub" to refer to an instance and not to the _service_.

I didn't have a strong opinion while coming up with this proposal, but now that I think about it, if I change the meaning to `destination: <type of instance>`, I will have to use `serviceBusQueue` and `serviceBusTopic` when I add support for Service Bus. But my original plan was to implement it as:

```yaml
destination:
  serviceBus:
    # stuff related to Service Bus: namespace, queue, topic, ...
```

Anyway, let me know if the intention is more clear after this explanation, and most importantly: whether you think I should revisit my original intention.

Reply:
Yeah it does seem like they use `Event Hubs` for the name of the technology and to encompass the namespace, but use `Event Hub` for the specific component. While looking at that, I thought destination was pointing to an instance of the service and not the service name itself.

I see where you're going with it, and it does make sense so I'll retract my objection.

Reply:
To provide more context, the spec of this source is meant to echo the direction I took with the Amazon S3 spec in #269, where `destination` is either `sqs` or `sns` (name of _services_) and the attributes related to the _instances_ within those services are nested (ARN, optional configs, API endpoints, etc.)

As long as things are consistent between specs of different source (current focus is on AWS, Azure and Google Cloud) and feel right for everyone, I'm OK with amending the original proposal.
We are exploring how to evolve the specs anyway, so no harm mentioning it when something feels off. It's still early enough to revisit!